internal/resources/google/kms_crypto_key.go

```diff
@@ -0,0 +1,200 @@
+package google
+
+import (
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/infracost/infracost/internal/usage"
+	"github.com/shopspring/decimal"
+)
+
+type KMSCryptoKey struct {
+	Address                         string
+	Region                          string
+	VersionTemplate                 string
+	VersionTemplate0Algorithm       string
+	VersionTemplate0ProtectionLevel string
+	RotationPeriod                  string
+	KeyVersions                     *int64 `infracost_usage:"key_versions"`
+	MonthlyKeyOperations            *int64 `infracost_usage:"monthly_key_operations"`
+}
+
+var KMSCryptoKeyUsageSchema = []*schema.UsageItem{{Key: "key_versions", ValueType: schema.Int64, DefaultValue: 0}, {Key: "monthly_key_operations", ValueType: schema.Int64, DefaultValue: 0}}
+
+func (r *KMSCryptoKey) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+func (r *KMSCryptoKey) BuildResource() *schema.Resource {
+	region := r.Region
+
+	algorithm := "GOOGLE_SYMMETRIC_ENCRYPTION"
+	protectionLevel := "SOFTWARE"
+
+	if r.VersionTemplate != "" {
+		algorithm = r.VersionTemplate0Algorithm
+		protectionLevel = r.VersionTemplate0ProtectionLevel
+	}
+
+	var monthlyKeys *decimal.Decimal
+	if r.KeyVersions != nil {
+		monthlyKeys = decimalPtr(decimal.NewFromInt(*r.KeyVersions))
+	} else if r.RotationPeriod != "" {
+		rotationPeriod := (r.RotationPeriod)
```

Comment:
This assignment looks like it can be removed.

Reply:
yeah, fixed